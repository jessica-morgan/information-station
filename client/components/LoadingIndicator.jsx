import React from 'react'

const LoadingIndicator = () => {
  return (
    <div>
      <img src="/loading.gif"></img>
    </div>
  )
}

export default LoadingIndicator
