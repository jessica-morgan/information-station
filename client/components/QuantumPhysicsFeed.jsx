import { useState, useEffect } from 'react'
import { getQuatumPhysicsNewsFeed } from '../api/newsApi'
import React, { useGlobal } from 'reactn'
import { selectTitle, currentTitleSelected } from '../utils'
import SingleNewsArticle from './SingleNewsArticle'
import LoadingIndicator from './LoadingIndicator'
import { TitlesContainer, NewsFeedTitles, H3, NewsTitleH2, Hr } from '../styles'

const QuantumPhysicsFeed = () => {
  const [allArticles, setArticles] = useState()

  useEffect(() => {
    const abortController = new AbortController()
    getQuatumPhysicsNewsFeed({ signal: abortController.signal })
      .then(articles => {
        setArticles({
          articles
        })
      })
    return () => {
      abortController.abort()
    }
  }, [])
  const [titleSelected, setTitleSelected] = useGlobal('titleSelected')
  const [categorySelected, setCategorySelected] = useGlobal('categorySelected')
  const [currentTitle, setCurrentTitle] = useGlobal('currentTitle')

  if (allArticles && categorySelected) {
    return <TitlesContainer>
      {allArticles.articles.map((article, idx) => {
        const formattedDate = article.publishedAt.substring(0, 10)
        return (
          <NewsFeedTitles key={idx}>
            <H3 key={article.publishedAt}>{formattedDate}</H3>
            <NewsTitleH2 style={{ textDecoration: 'none' }} key={article.title} onClick={() => { selectTitle(true); currentTitleSelected(article.title) }}>{article.title}</NewsTitleH2>
            <H3 style={{ color: '#808080' }}>{article.description}</H3>
          </NewsFeedTitles>
        )
      })}
    </TitlesContainer>
  } if (allArticles && titleSelected) {
    return <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
      <SingleNewsArticle articles={allArticles.articles} key={currentTitle} />
    </div>
  } else {
    return <div>
      <LoadingIndicator />
    </div>
  }
}

export default QuantumPhysicsFeed
