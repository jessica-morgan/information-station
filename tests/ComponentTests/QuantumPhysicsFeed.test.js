import React from 'reactn'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import QuantumPhysicsFeed from '../../client/components/QuantumPhysicsFeed'
import { mount } from 'enzyme'
import LoadingIndicator from '../../client/components/LoadingIndicator'

test('IsItBullshit renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <QuantumPhysicsFeed />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('IsItBullshit should render LoadingIndicator if posts is not yet set with api data', done => {
  const wrapper = mount(<QuantumPhysicsFeed />)
  setImmediate(() => {
    wrapper.update()
    expect(wrapper.containsMatchingElement(<LoadingIndicator></LoadingIndicator>)).toBeTruthy()
    done()
  })
})