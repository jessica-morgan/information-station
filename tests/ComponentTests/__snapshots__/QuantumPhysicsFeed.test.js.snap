// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantumPhysicsFeed renders correctly 1`] = `
<div>
  <div>
    <img
      className="loading-indicator"
      src="/loading.gif"
    />
  </div>
</div>
`;
